src/TypesService.ts

```diff
@@ -82,9 +82,17 @@ export const mapFileAndExtension: Endomorphism<Endomorphism<string>> =
     return rewrittenPath.startsWith('.') ? rewrittenPath : `./${rewrittenPath}`
   }
 
+const getSourceFile = (node: ts.Node): string =>
+  node.getSourceFile()?.fileName ??
+  // This isn't made explicit in the types, but Typescript will add an
+  // `original` property to nodes when they get `update`d, and their
+  // `getSourceFile()` returns undefined. This is a workaround for that.
+  (node as typeof node & { original?: typeof node }).original?.getSourceFile()?.fileName
```

Comment:
Here you could change `typeof node` to `ts.Node`

Reply:
My thinking was that the node type might already have been narrowed. But this is in a function where the type is explicitly just `ts.Node`. So.. yeah. Will change